Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -30,85 +27,240 @@ export default function widgetBehavior(publicAPI, model) {
+    model.widgetState.getText().setText(text);
+    model.interactor.render();
+  };
+
+  // --------------------------------------------------------------------------
+  // Handle positioning methods
+  // --------------------------------------------------------------------------
+
+  // Handle utilities ---------------------------------------------------------
+
+  function computeDirectionWithTwoPoints(p1, p2) {
+    const dir = [0, 0, 0];
+    vtkMath.subtract(p1, p2, dir);
+    return dir;
   }
 
   function isHandleOrientable(handleType) {
-    if (
+    return (
       handleType === HandleRepresentationType.CONE ||
       handleType === HandleRepresentationType.ARROWHEAD3 ||
       handleType === HandleRepresentationType.ARROWHEAD4 ||
       handleType === HandleRepresentationType.ARROWHEAD6
-    )
-      return 1;
-    return 0;
+    );
   }
 
   function isOrientable() {
     return (
-      isHandleOrientable(model.handle1Shape) ||
-      isHandleOrientable(model.handle2Shape)
+      isHandleOrientable(handle1.getShape()) ||
+      isHandleOrientable(handle2.getShape())
     );
   }
 
-  // set in public to update handle  Direction when handle change in UI
-  publicAPI.setHandleDirection = () => {
-    if (isHandleOrientable(model.handle1Shape)) {
-      updateHandleDirection(HandleBehavior.HANDLE1);
+  function computeMousePosition(p1, callData) {
+    const worldMousePos = publicAPI.computeWorldToDisplay(
+      model.renderer,
+      p1[0],
+      p1[1],
+      p1[2]
+    );
+    const mousePos = publicAPI.computeDisplayToWorld(
+      model.renderer,
+      callData.position.x,
+      callData.position.y,
+      worldMousePos[2]
+    );
+    return mousePos;
+  }
+
+  publicAPI.isHandleGlyph2D = (handleShape) =>
+    handleShape === HandleRepresentationType.ARROWHEAD3 ||
+    handleShape === HandleRepresentationType.ARROWHEAD4 ||
+    handleShape === HandleRepresentationType.ARROWHEAD6 ||
+    handleShape === HandleRepresentationType.STAR ||
+    handleShape === HandleRepresentationType.CIRCLE ||
+    handleShape === HandleRepresentationType.VIEWFINDER ||
+    handleShape === HandleRepresentationType.DISK;
+
+  // Handle orientation & rotation ---------------------------------------------------------
+
+  function orientFirstHandleBeforeSecondHandlePlacement(callData) {
+    const handle1Origin = handle1.getOrigin();
+    const mousePos = computeMousePosition(handle1Origin, callData);
+    return computeDirectionWithTwoPoints(handle1Origin, mousePos);
+  }
+
+  function orientHandlesWithCompleteWidget(nb) {
+    let dir = [0, 0, 0];
+    const handle1Origin = handle1.getOrigin();
+    const handle2Origin =
+      nb !== 3
+        ? handle2.getOrigin()
+        : model.widgetState.getMoveHandle().getOrigin();
+    dir = computeDirectionWithTwoPoints(handle1Origin, handle2Origin);
+    if (nb === 2 || nb === 3) {
+      vtkMath.multiplyScalar(dir, -1);
+    }
+    return dir;
+  }
+
+  function orientHandle(nb, direction) {
```

Comment:
this `nb` parameter is strange. 
Please add function documentation.
It's not easy to read, because you update the handle `nb-1` and not `nb`.
It seems to me that it is more a `handleIndex` that this function must take as parameter rather than a pseudo "number of handles"  (which is different from `getNbHandles`)